homeassistant/components/blueprint/importer.py

```diff
@@ -245,13 +245,32 @@ async def fetch_blueprint_from_website_url(
     return ImportedBlueprint(suggested_filename, raw_yaml, blueprint)
 
 
+async def fetch_blueprint_from_generic_url(
+    hass: HomeAssistant, url: str
+) -> ImportedBlueprint:
+    """Get a blueprint from a generic website."""
+    session = aiohttp_client.async_get_clientsession(hass)
+
+    resp = await session.get(url, raise_for_status=True)
+    raw_yaml = await resp.text()
+    data = yaml.parse_yaml(raw_yaml)
+
+    assert isinstance(data, dict)
```

Comment:
The error handling is not great, but it's obviously copied from the other importers.

In a separate PR, I think we should replace the assert with raising a HomeAssistantError if `data` is not a dict?

Reply:
Do you want this separate PR to be done before this one gets merged, or should I do it afterwards?

Reply:
Let's improve the error handling in another PR then